### ISSUE TITLE
performance did not improve when i increase the number of processes

### ISSUE THREAD

**Issue by zly9844** [horovod/horovod#3785]
I want to know the differences on performance between using horovod and without horovod, so i use a pytorch version of mnist demo with different settings. Here's the result:

 1. without hovorod, batch_size = 32
 
	Test set: Average loss: 0.0466, Accuracy: 98.62%
	duration:  0:03:23.520224
 	
 2. with horovod, np = 1, batch_size = 32
 
	Test set: Average loss: 0.0425, Accuracy: 98.64%
	duration:  0:03:09.118766
 
 3. with horovod, np = 2, batch_size = 32 * 2
 
	Test set: Average loss: 0.0535, Accuracy: 98.30%
	duration:  0:02:13.352439

 4. with horovod, np = 4, batch_size = 32 * 4
 
	Test set: Average loss: 0.0717, Accuracy: 97.84%
	duration:  0:02:42.104210

All the other settings are the same, from the results above, the training speed has an increase when using horovod compared with not using horovod. 

But what puzzles me is that when i increase the number of processes from 2 to 4, the performance becomes worse. Is that normal? Or maybe there are some other settings that I ignore?
